.github/workflows/simulate-merge-pr.yml

```diff
@@ -0,0 +1,95 @@
+name: Simulate merging a PR 
```

Comment:
🤔 feels like this file and `test-deployment.yml` could be merged. I don't imagine we'd run the `test-deployment` workflow separately.

If so, I would suggest renaming the file to `simulate-release.yml`.

What do you think?